packages/gestalt/src/IconButtonFloating.js

```diff
@@ -0,0 +1,69 @@
+// @flow strict
+import { forwardRef, type Node, useImperativeHandle, useRef } from 'react';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+import icons from './icons/index.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
+
+type Props = {|
+  /**
+   * String that clients such as VoiceOver will read to describe the icon button. Always localize the label. See [Accessibility section](https://gestalt.pinterest.systems/web/modalalert#Accessibility) for more info.
+   */
+  accessibilityLabel: string,
+  icon?: $Keys<typeof icons>,
+  onClick?: AbstractEventHandler<
+    | SyntheticMouseEvent<HTMLButtonElement>
+    | SyntheticKeyboardEvent<HTMLButtonElement>
+    | SyntheticMouseEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    {| dangerouslyDisableOnNavigation: () => void |},
+  >,
+  accessibilityControls?: string,
+  accessibilityExpanded?: boolean,
+  accessibilityHaspopup?: boolean,
+|};
+
+/**
+ * A [IconButtonFloating](https://gestalt.pinterest.systems/web/iconbuttonfloating) represents the primary or most common action on the screen. As the name suggests, it floats over the content and is always on top of everything on the screen. Similar to the IconButton, the floating version uses icons instead of text to convey available actions; however, it is used when the action needs to be visible at all times in a sticky way where content can scroll underneath. IconButtonFloating remains in place on scroll.
+
+By default, it has a circular shape with a floating elevation shadow style built-in; when pressed, it will open more related actions by triggering a Dropdown or a Modal.
+
+IconButtonFloating is typically found in the Home feed, boards, and dashboards, supporting Pinners to perform core actions.
+ */
+const IconButtonFloatingWithForwardRef: React$AbstractComponent<Props, HTMLButtonElement> =
+  forwardRef<Props, HTMLButtonElement>(function IconButtonFloating(props: Props, ref): Node {
```

Comment:
cleanup

as a pattern, we destructure inside the function 

IconButtonFloating({ 
a,
b,
c
}: Props, ref)